src/Compiler/Symbols/FSharpDiagnostic.fs

```diff
@@ -319,23 +318,18 @@ type internal CompilationDiagnosticLogger (debugName: string, options: FSharpDia
 module DiagnosticHelpers =                            
 
     let ReportDiagnostic (options: FSharpDiagnosticOptions, allErrors, mainInputFileName, fileInfo, diagnostic: PhasedDiagnostic, severity, suggestNames, flatErrors, symbolEnv) =
-        [ let severity = 
-               if diagnostic.ReportAsError (options, severity) then
-                   FSharpDiagnosticSeverity.Error
-               else
-                   severity
-
-          if severity = FSharpDiagnosticSeverity.Error ||
-             diagnostic.ReportAsWarning (options, severity) ||
-             diagnostic.ReportAsInfo (options, severity) then 
+        match diagnostic.AdjustedSeverity(options, severity) with
+        | FSharpDiagnosticSeverity.Hidden -> []
+        | sev ->
 
             // We use the first line of the file as a fallbackRange for reporting unexpected errors.
             // Not ideal, but it's hard to see what else to do.
             let fallbackRange = rangeN mainInputFileName 1
-            let diagnostic = FSharpDiagnostic.CreateFromExceptionAndAdjustEof (diagnostic, severity, fallbackRange, fileInfo, suggestNames, flatErrors, symbolEnv)
+            let diagnostic = FSharpDiagnostic.CreateFromExceptionAndAdjustEof (diagnostic, sev, fallbackRange, fileInfo, suggestNames, flatErrors, symbolEnv)
```

Comment:
It was originally severity, we have always had problems with abbreviations in this codebase, in general avoid them, especially if they were originally unabbreviated.

Reply:
I didn't want to shadow the original severity. I'll make it `adjustedSeverity`.